examples/hello-recursion.yaml

```diff
@@ -0,0 +1,43 @@
+# The following workflow executes a workflow which the number of next level depends on the output of the previous step.
+# A is the first step. After a is executed, b is executed (b generates n b based on the result of a),
+# and after b is executed, c is executed (c generates n c based on the result of each b).
```

Comment:
```suggestion
# A is the first step. A's output determines how many times B is executed.
# B's output then determines how many times C is executed.
```

- more simple and direct, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
  - no parentheses, no "n"
- consistently use the same capitalization (capital A, B, C)
- fix run-on sentence